task2_Gal_Malka/include/ChessPiece.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include "Position.h"
+
+enum Color : bool
+{
+	White = true,
+	Black = false
+};
+
+enum class MaxDistance {
+	OneSquare = 1,
+	TwoSquares,
+	ThreeSquares,
+	FullBoard = 18
+};
+
+class ChessPiece : public virtual IMovement{
```

Comment:
You should mark this class as abstract
https://stackoverflow.com/questions/43502488/declare-abstract-class-in-c